subsys/nrf_security/src/drivers/cracen/psa_driver.Kconfig

```diff
@@ -12,380 +12,719 @@ visible if 0
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_DERIVE_MONTGOMERY_448
+	select PSA_ACCEL_KEY_TYPE_ECC_PUBLIC_KEY_MONTGOMERY_448
+	# Brainpool 256
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_IMPORT_BRAINPOOL_P_R1_256
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_EXPORT_BRAINPOOL_P_R1_256
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_GENERATE_BRAINPOOL_P_R1_256
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_DERIVE_BRAINPOOL_P_R1_256
+	select PSA_ACCEL_KEY_TYPE_ECC_PUBLIC_KEY_BRAINPOOL_P_R1_256
+	# Brainpool 384
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_IMPORT_BRAINPOOL_P_R1_384
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_EXPORT_BRAINPOOL_P_R1_384
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_GENERATE_BRAINPOOL_P_R1_384
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_DERIVE_BRAINPOOL_P_R1_384
+	select PSA_ACCEL_KEY_TYPE_ECC_PUBLIC_KEY_BRAINPOOL_P_R1_384
+	# Brainpool 512
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_IMPORT_BRAINPOOL_P_R1_512
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_EXPORT_BRAINPOOL_P_R1_512
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_GENERATE_BRAINPOOL_P_R1_512
+	select PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR_DERIVE_BRAINPOOL_P_R1_512
+	select PSA_ACCEL_KEY_TYPE_ECC_PUBLIC_KEY_BRAINPOOL_P_R1_512
+	# Depends
+	depends on PSA_WANT_ECC_SECP_R1_192      	|| \
+		   PSA_WANT_ECC_SECP_R1_224         	|| \
+		   PSA_WANT_ECC_SECP_R1_256         	|| \
+		   PSA_WANT_ECC_SECP_R1_384         	|| \
+		   PSA_WANT_ECC_SECP_K1_192         	|| \
+		   PSA_WANT_ECC_SECP_K1_224         	|| \
+		   PSA_WANT_ECC_SECP_K1_256         	|| \
+		   PSA_WANT_ECC_TWISTED_EDWARDS_255 	|| \
+		   PSA_WANT_ECC_TWISTED_EDWARDS_448 	|| \
+		   PSA_WANT_ECC_MONTGOMERY_255      	|| \
+		   PSA_WANT_ECC_MONTGOMERY_448      	|| \
+		   PSA_WANT_ECC_BRAINPOOL_P_R1_256  	|| \
+		   PSA_WANT_ECC_BRAINPOOL_P_R1_384  	|| \
+		   PSA_WANT_ECC_BRAINPOOL_P_R1_512
+	depends on PSA_USE_CRACEN_KEY_MANAGEMENT_DRIVER
 
 config PSA_NEED_CRACEN_KEY_MANAGEMENT_DRIVER
 	bool
 	default y
+	depends on PSA_NEED_CRACEN_KEY_TYPE_RSA_ANY || \
+	           PSA_NEED_CRACEN_KEY_TYPE_ECC_ANY
 
 # CRACEN MAC Driver
 
 config PSA_NEED_CRACEN_HMAC
 	bool
 	default y
+	select PSA_ACCEL_HMAC_SHA_1   if PSA_WANT_ALG_SHA_1
+	select PSA_ACCEL_HMAC_SHA_224 if PSA_WANT_ALG_SHA_224
+	select PSA_ACCEL_HMAC_SHA_256 if PSA_WANT_ALG_SHA_256
+	select PSA_ACCEL_HMAC_SHA_384 if PSA_WANT_ALG_SHA_384
+	select PSA_ACCEL_HMAC_SHA_512 if PSA_WANT_ALG_SHA_512
+	depends on PSA_WANT_ALG_HMAC
+	depends on PSA_USE_CRACEN_MAC_DRIVER
 
 config PSA_NEED_CRACEN_CMAC
 	bool
 	default y
+	select PSA_ACCEL_CMAC_AES_128
+	select PSA_ACCEL_CMAC_AES_192
+	select PSA_ACCEL_CMAC_AES_256
+	depends on PSA_WANT_ALG_CMAC
+	depends on PSA_USE_CRACEN_MAC_DRIVER
 
 config PSA_NEED_CRACEN_MAC_DRIVER
 	bool
 	default y
+	depends on PSA_NEED_CRACEN_HMAC || PSA_NEED_CRACEN_CMAC
 
 # CRACEN Entropy Driver
 
 config PSA_NEED_CRACEN_CTR_DRBG_DRIVER
 	bool
 	default y
+	select PSA_ACCEL_GENERATE_RANDOM
+	# Is needed for AES counter measurements
+	depends on (PSA_WANT_ALG_CTR_DRBG		&& \
```

Comment:
Don't we want to be using CTR_DRBG for AES countermeasures independently of whether the user has asked for CTR_DRBG?

Reply:
Yes it gets selected if the user wants it `(PSA_WANT_ALG_CTR_DRBG && PSA_USE_CRACEN_CTR_DRBG_DRIVER)`
or if we use cryptomaster
`PSA_USE_CRACEN_CIPHER_DRIVER || PSA_USE_CRACEN_AEAD_DRIVER || PSA_USE_CRACEN_MAC_DRIVER`

Reply:
ah, now I got it, nicely done :)